src/main/scala/elevate/rise/strategies/predicate.scala

```diff
@@ -28,11 +29,28 @@ object predicate {
   case object isReduce extends Strategy[Rise] {
     def apply(e: Rise): RewriteResult[Rise] = e match {
       case r@Reduce() => Success(r)
-      case _          => Failure(isMap)
+      case _          => Failure(isReduce)
     }
     override def toString = "isReduce"
   }
 
+  case object isReduceSeq extends Strategy[Rise] {
+    def apply(e: Rise): RewriteResult[Rise] = e match {
+      case r@ReduceSeq() => Success(r)
+      case _             => Failure(isReduceSeq)
+    }
+    override def toString = "isReduce"
+  }
+
+  def isReduceX: Strategy[Rise] = (isReduce <+ isReduceSeq)
```

Comment:
I think this is defined elsewhere also?

Reply:
it was but we removed it once we no longer needed all the extractors to match TypedExpr. this is the only definition of `isReduceX`